pkg/validator/validator_test.go

```diff
@@ -5,33 +5,32 @@ import (
 )
 
 var testData = []struct {
-	name string
-	testInput []byte
+	name           string
+	testInput      []byte
 	expectedResult bool
-	validator Validator
+	validator      Validator
 }{
-  {"validJson", []byte(`{"test": "test"}`), true, JsonValidator{}},
-  {"invalidJson", []byte(`{test": "test"}`), false, JsonValidator{}},
-  {"validYaml", []byte("a: 1\nb: 2"), true, YamlValidator{}},
-  {"invalidYaml", []byte("a: b\nc: d:::::::::::::::"), false, YamlValidator{}},
-  {"validXml", []byte("<test>\n</test>"), true, XmlValidator{}},
-  {"invalidXml", []byte("<xml\n"), false, XmlValidator{}},
+	{"validJson", []byte(`{"test": "test"}`), true, JsonValidator{}},
+	{"invalidJson", []byte(`{test": "test"}`), false, JsonValidator{}},
+	{"validYaml", []byte("a: 1\nb: 2"), true, YamlValidator{}},
+	{"invalidYaml", []byte("a: b\nc: d:::::::::::::::"), false, YamlValidator{}},
+	{"validXml", []byte("<test>\n</test>"), true, XmlValidator{}},
+	{"invalidXml", []byte("<xml\n"), false, XmlValidator{}},
 }
 
-
 func Test_ValidationInput(t *testing.T) {
 	for _, d := range testData {
-			valid, err := d.validator.Validate(d.testInput)
-			if valid != d.expectedResult {
-				t.Errorf("incorrect result: expected %v, got %v", d.expectedResult, valid)
-			}
-	
-			if valid && err != nil {
-				t.Error("incorrect result: err was not nil", err)
-			}
+		valid, err := d.validator.Validate(d.testInput)
+		if valid != d.expectedResult {
+			t.Errorf("incorrect result: expected %v, got %v", d.expectedResult, valid)
+		}
 
-			if ! valid && err == nil {
-				t.Error("incorrect result: function returned a nil error")
-			}
+		if valid && err != nil {
+			t.Error("incorrect result: err was not nil", err)
 		}
-}
\ No newline at end of file
+
+		if !valid && err == nil {
```

Comment:
Is there a way to combine these with `if !valid || err != nil` and then just make the error a little more generic by formatting a string with valid and err

Reply:
There's probably a better way to do it than is currently implemented but combining them into `if !valid || err != nil` doesn't quite test the same thing. The first if statement checks that an error is not returned if the function returns true and the second ensures that an error is returned if the function returns false. The test should still pass if the function returns false if that's the expected result.

Reply:
Yeah sorry that should be an xor but I just realized go doesn't have that builtin so it wouldn't make it prettier anyways. Oh well.